tls/s2n_kex.c

```diff
@@ -41,17 +41,17 @@ static int s2n_check_ecdhe(const struct s2n_cipher_suite *cipher_suite, struct s
 
 static int s2n_check_kem(const struct s2n_cipher_suite *cipher_suite, struct s2n_connection *conn)
 {
-    notnull_check(conn);
-    /* There is no support for PQ KEMs while in FIPS mode */
-    if (s2n_is_in_fips_mode()) {
+    /* The return value is treated like a bool, so we avoid using GUARD, notnull_check, etc. */
+    if (conn == NULL || cipher_suite == NULL) {
         return 0;
     }
 
     const struct s2n_kem_preferences *kem_preferences = NULL;
-    GUARD(s2n_connection_get_kem_preferences(conn, &kem_preferences));
-    notnull_check(kem_preferences);
+    if (s2n_connection_get_kem_preferences(conn, &kem_preferences) != S2N_SUCCESS) {
+        return 0;
```

Comment:
Nit: While you're already making changes to this file unrelated to the consolidated PQ check, you might want to change all these return types to S2N_SUCCESS.